client/fingerprint/env_aws.go

```diff
@@ -156,6 +530,28 @@ func (f *EnvAWSFingerprint) Fingerprint(request *FingerprintRequest, response *F
 	return nil
 }
 
+func (f *EnvAWSFingerprint) instanceType(ec2meta *ec2metadata.EC2Metadata) (string, error) {
+	response, err := ec2meta.GetMetadata("instance-type")
+	if err != nil {
+		return "", err
+	}
+	return strings.TrimSpace(response), nil
+}
+
+func (f *EnvAWSFingerprint) lookupCPU(ec2meta *ec2metadata.EC2Metadata) *ec2Specs {
+	instanceType, err := f.instanceType(ec2meta)
+	if err != nil {
+		f.logger.Error("error reading instance-type", "error", err)
```

Comment:
I would downgrade the log to warning.  It's not an actionable for operators, and it's possible they are running on environments that mimic some EC2 metadata but don't show instance-type.

```suggestion
		f.logger.Warn("failed to read EC2 metadata instance-type", "error", err)
```